### DISCUSSION TITLE
Cannot delete or update a parent row

### DISCUSSION THREAD

**Discussion by LeadcodeDev** [adonisjs/core#1573] [Help] [answered]
Hi !

I would like to delete one service but when I use delete method on my controller, I have this error
```
[1599566990303] ERROR (api/16756 on DESKTOP-9Q3Q14G): delete from `services` where `id` = 2 - ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`portfolio-adonis-nuxt`.`details`, CONSTRAINT `details_service_id_foreign` FOREIGN KEY (`service_id`) REFERENCES `services` (`id`))
```

Following my service model :
```ts
export default class Service extends BaseModel {
	@column({ isPrimary: true })
	public id: number

	@column()
	public label: string

	@column()
	public description: string

	@column()
	public isVisible: boolean

	@hasMany(() => Detail)
	public details: HasMany<typeof Detail>

	@hasOne(() => Picture)
	public picture: HasOne<typeof Picture>
}
```

How can I delete my item while also deleting the assigned relationships ?

**Comment by RomainLanz** [comment, marked as answer]
Hey @LeadcodeDev! 👋 

You need to properly create your table.
When creating the foreign key in your migration, you should specify using the [`onDelete`](https://knexjs.org/#Schema-onDelete) hook what you want to do. For your case, it should be `cascade`.

**Reply by LeadcodeDev** [reply]
It seemed to me that I had specified the `onDelete ('CASCADE')` hook in my migration....
After checking, I had forgotten it 😅

I put my migration in this post for future people who will encounter the problem :
```ts
public async up() {
	this.schema.createTable(this.tableName, (table) => {
		table.increments('id')
		table.string('description')
		table
			.integer('service_id')
			.unsigned()
			.references('id')
			.inTable('services')
			.onDelete('CASCADE')
		table.timestamps(true, true)
	})
}
```
